File: 83.删除排序链表中的重复元素.py
#
# @lc app=leetcode.cn id=83 lang=python3
#
# [83] 删除排序链表中的重复元素
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        dummy_node = head
        if not head:
            return head
        pre_node = head
        head = head.next
        while head:
            if head.val == pre_node.val:
                pre_node.next = head.next
            else:
                pre_node = head
            head = head.next
        return dummy_node
# @lc code=end

File: 83.删除排序链表中的重复元素.go
/*
 * @lc app=leetcode.cn id=83 lang=golang
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func deleteDuplicates(head *ListNode) *ListNode {
    dummyNode := head
    preNode := head
    if head == nil {
        return head
    } else {
        head = head.Next
    }
    for head != nil {
        if head.Val == preNode.Val {
            // 存在重复元素，删除
            preNode.Next = head.Next
        } else {
            preNode = head
        }
        head = head.Next
    }
    return dummyNode
}
// @lc code=end

